build_tools/cmake/iree_installed_test.cmake

```diff
@@ -0,0 +1,77 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# iree_add_installed_test()
+function(iree_add_installed_test)
+  cmake_parse_arguments(
+    _RULE
+    ""
+    "NAME"
+    "COMMAND;ENVIRONMENT;INSTALLED_COMMAND;CONFIGURATIONS;WORKING_DIRECTORY;LABELS"
+    ${ARGN}
+  )
+
+  # Prefix the library with the package name, so we get: iree_package_name
+  iree_package_name(_PACKAGE_NAME)
+  set(_NAME "${_PACKAGE_NAME}_${_RULE_NAME}")
+
+  # Replace dependencies passed by ::name with iree::package::name
+  iree_package_ns(_PACKAGE_NS)
+  list(TRANSFORM _RULE_DEPS REPLACE "^::" "${_PACKAGE_NS}::")
+  string(REPLACE "::" "/" _PACKAGE_PATH ${_PACKAGE_NS})
+  set(_TEST_NAME "${_PACKAGE_PATH}/${_RULE_NAME}")
+
+  add_test(
+    NAME ${_TEST_NAME}
```

Comment:
Can we make the API for this just pass the test name rather than redoing this computation?

Reply:
Oh, good call. This was more intertwined but now it isn't and is easily separable.